src/components/button-group/button-group.ts

```diff
@@ -0,0 +1,288 @@
+  private toggleButtons!: Array<IgcToggleButtonComponent>;
+
+  /**
+   * Disables all buttons inside the group.
+   * @attr
+   */
+  @property({ type: Boolean, reflect: true })
+  public disabled = false;
+
+  /**
+   * Sets the orientation of the buttons in the group.
+   * @attr
+   */
+  @property({ reflect: true })
+  public alignment: 'horizontal' | 'vertical' = 'horizontal';
+
+  /**
+   * Controls the mode of selection for the button group.
+   * @attr
+   */
+  @property({ reflect: false })
+  public selection: 'single' | 'single-required' | 'multiple' = 'single';
+
+  /**
+   * Gets/Sets the currently selected buttons (their values).
+   * @attr
+   */
+  @property({ attribute: 'selected-items', type: Array, reflect: false })
+  public get selectedItems() {
+    return this._selectedButtons.map((b) => b.value).filter((v) => v);
+  }
+
+  public set selectedItems(values: string[]) {
+    if (values && values.length) {
+      this._selectedItems = values;
+      this.setSelection(values);
+    }
+  }
+
+  @watch('disabled', { waitUntilFirstUpdate: true })
+  protected updateDisabledState() {
+    this.toggleButtons.forEach((b) => (b.disabled = this.disabled));
+  }
+
+  @watch('selection', { waitUntilFirstUpdate: true })
+  protected updateSelectionState() {
+    if (this._selectedButtons.length) {
+      this.toggleButtons.forEach((b) => (b.selected = false));
+    }
+  }
+
+  public override connectedCallback() {
+    super.connectedCallback();
+    this.id =
+      this.getAttribute('id') ||
+      `igc-button-group-${IgcButtonGroupComponent.increment()}`;
+  }
+
+  public override disconnectedCallback() {
+    super.disconnectedCallback();
+    this.mutationObserver.disconnect();
+  }
+
+  protected override firstUpdated() {
+    if (this.disabled) {
+      this.updateDisabledState();
+    }
+
+    // Button level selection takes priority
+    if (this._selectedItems.length && !this._selectedButtons.length) {
+      this.setSelection(this._selectedItems);
+    } else if (
+      this.selection !== 'multiple' &&
+      this._selectedButtons.length > 1
+    ) {
+      const index = this._selectedButtons.indexOf(
+        this._selectedButtons.at(-1) as IgcToggleButtonComponent
+      );
+
+      for (let i = 0; i < index; i++) {
+        this._selectedButtons[i].selected = false;
+      }
+    }
+
+    this.setMutationsObserver();
+  }
+
+  private handleClick = (event: MouseEvent) => {
+    const button = event
+      .composedPath()
+      .find(
+        (element) => element instanceof IgcToggleButtonComponent
+      ) as IgcToggleButtonComponent;
+
+    if (button) {
+      switch (this.selection) {
+        case 'single':
+          this.handleSingleSelection(button);
+          break;
+        case 'single-required':
+          this.handleSingleSelection(button, true);
+          break;
+        case 'multiple':
+          this.handleMultipleSelection(button);
+          break;
+      }
+    }
+  };
```

Comment:
```suggestion
  private handleClick(event: MouseEvent) {
    const button = event
      .composedPath()
      .find(
        (element) => element instanceof IgcToggleButtonComponent
      ) as IgcToggleButtonComponent;

    if (button) {
      this.selection === 'multiple'
        ? this.handleMultipleSelection(button)
        : this.handleSingleSelection(button);
    }
  }
```
There is no need to have the handler as an arrow bound function.
The logic for single selection can also be moved inside the `handleSingleSelection()`.